src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,268 @@
+                )
+            )
+        sdk_zip_path.unlink()  # Zip file no longer needed once unpacked.
+        for binpath in tools_path.glob("*"):
+            binpath.chmod(0o755)
+
+    def verify_license(self):
+        license_path = self.sdk_path / "licenses" / "android-sdk-license"
+        if license_path.exists():
+            return
+
+        print("\nPlease accept `android-sdk-license` when prompted.\n")
+        try:
+            self.subprocess.run(
+                ["./sdkmanager", "--licenses"],
+                check=True,
+                cwd=str(self.sdk_path / "tools" / "bin"),
+            )
+        except subprocess.CalledProcessError:
+            raise BriefcaseCommandError(
+                """\
+Error while reviewing Android SDK licenses. Please run this command and examine
+its output for errors.
+
+$ {sdkmanager} --licenses""".format(
+                    sdkmanager=self.sdk_path / "tools" / "bin" / "sdkmanager"
+                )
+            )
+
+        if not license_path.exists():
+            raise BriefcaseCommandError(
+                """\
+You did not accept the Android SDK license. Please re-run the briefcase command
+and accept the Android SDK license when prompted. You may need an Internet
+connection."""
+            )
+
+    def verify_tools(self):
+        """
+        Verify that we the Android APK tools in `briefcase` will operate on
+        this system, downloading tools as needed.
+        """
+        self.verify_python_version()
+        self.verify_sdk()
+        self.verify_license()
+
+
+class ApkCreateCommand(ApkMixin, CreateCommand):
+    description = "Create and populate an Android APK."
+
+
+class ApkUpdateCommand(ApkMixin, UpdateCommand):
+    description = "Update an existing Android APK."
+
+
+class ApkBuildCommand(ApkMixin, BuildCommand):
+    description = "Build an Android APK."
+
+    def build_app(self, app: BaseConfig, **kwargs):
+        """
+        Build an application.
+
+        :param app: The application to build
+        """
+        print("[{app.app_name}] Building Android APK...".format(app=app))
+        try:
+            env = {**self.os.environ, "ANDROID_SDK_ROOT": str(self.sdk_path)}
+            self.subprocess.check_output(
+                ["./gradlew", "assembleDebug"],
+                env=env,
+                cwd=str(self.bundle_path(app)),
+                stderr=self.subprocess.STDOUT,
+            )
+        except subprocess.CalledProcessError as e:
+            raise BriefcaseCommandError(
+                "Error while building. Full gradle output:\n\n"
+                + e.output.decode("ascii", "replace")
+            )
+
+
+class ApkRunCommand(ApkMixin, RunCommand):
+    description = "Run an Android APK."
+
+    def verify_tools(self):
+        super().verify_tools()
+        if (self.sdk_path / "emulator").exists():
+            return
+
+        print("Ensuring we have the Android emulator and system image...")
```

Comment:
We've already ensured... this is "Downloading"

Reply:
+1

Reply:
Resolved in latest push